src/mon/MDSMonitor.cc

```diff
@@ -146,7 +146,12 @@ void MDSMonitor::update_from_paxos(bool *need_bootstrap)
 
   ceph_assert(fsmap_bl.length() > 0);
   dout(10) << __func__ << " got " << version << dendl;
-  PaxosFSMap::decode(fsmap_bl);
+  try {
+    PaxosFSMap::decode(fsmap_bl);
+  } catch (const ceph::buffer::malformed_input& e) {
+    derr << "FSMap is too old or malformed" << dendl;
```

Comment:
nit, could print more info, like
```c++
derr << "FSMap is too old or malformed: " << e.what() << dendl;
```
so we can at least tell if blob is encoded with an ancient scheme or it is just corrupted. because `DECODE_START()` raises `::ceph::buffer::malformed_input(DECODE_ERR_OLDVERSION(__PRETTY_FUNCTION__, v, struct_compat))` at seeing an unsupported `struct_compat` version.

Reply:
Thanks, done.